test/test_respondent_controller.py

```diff
@@ -2032,3 +2035,84 @@ def _enroll_respondent(self, session):
+    def test_all_respondents_marked_are_delete(self, mock_session, mock_send_account_deletion_confirmation_email):
+        respondent_1 = Respondent()
+        respondent_1.email_address = "test1@example.com"
+        respondent_1.first_name = "One"
+
+        respondent_2 = Respondent()
+        respondent_2.email_address = "test2@example.com"
+        respondent_2.first_name = "Two"
+
+        respondent_3 = Respondent()
+        respondent_3.email_address = "test3@example.com"
+        respondent_3.first_name = "Three"
+
+        respondents_to_delete = [respondent_1, respondent_2, respondent_3]
+
+        mock_query = MagicMock()
+        mock_query.filter(Respondent.mark_for_deletion).return_value = respondents_to_delete
+        # To iterate over "for respondent in respondents" the __iter__ method needs defining for the Mock
+        mock_query.filter.return_value.__iter__.return_value = iter(respondents_to_delete)
+        mock_session.query.return_value = mock_query
+
+        # Execute
+        with self.app.app_context():
+            delete_respondents_marked_for_deletion.__wrapped__(mock_session)
+
+        # Verify
+        expected_confirmation_email_calls = [
+            call(respondent_1.email_address, respondent_1.first_name),
+            call(respondent_2.email_address, respondent_2.first_name),
+            call(respondent_3.email_address, respondent_3.first_name),
+        ]
+        mock_send_account_deletion_confirmation_email.assert_has_calls(expected_confirmation_email_calls)
+        self.assertEqual(mock_session.commit.call_count, 3)
+        mock_session.rollback.assert_not_called()
+
+    @patch("ras_party.controllers.respondent_controller.delete_respondent_records")
+    @patch("ras_party.controllers.respondent_controller.send_account_deletion_confirmation_email")
+    @patch("ras_party.controllers.respondent_controller.session", new_callable=MagicMock)
+    def test_delete_respondents_continues_when_exception(
+        self, mock_session, mock_send_account_deletion_confirmation_email, mock_delete_respondent_records
+    ):
+        # Setup
+        respondent_1 = Respondent()
+        respondent_1.email_address = "test1@example.com"
+        respondent_1.first_name = "One"
+
+        respondent_2 = Respondent()
+        respondent_2.email_address = "test2@example.com"
+        respondent_2.first_name = "Two"
+
+        respondent_3 = Respondent()
+        respondent_3.email_address = "test3@example.com"
+        respondent_3.first_name = "Three"
+
+        respondents_to_delete = [respondent_1, respondent_2, respondent_3]
+
+        # Ideally this test would only throw an exception for respondent_2
+        # so respondent_1 and respondent_3 would be committed and respondent_2 would be rolled back
+        # but I can't work out how to do that yet so we're throwing exceptions for all 3
+        # this is good enough to test the production bug fix as previously it was terminating after a single exception
```

Comment:
I can't find a way to conditionally raise an exception in the class under test ONLY for respondent_2 so I just mock raising an exception for all three. Ideally I'd want to assert that two respondents `commit()` and one respondent `rollback()` showing that the process continues. This test does indeed show the process continues, but it does that while raising exceptions for every respondent under test (which may indeed happen so its still a valid test)